extension/src/Dictionary.ts

```diff
@@ -0,0 +1,165 @@
+import { existsSync, readFileSync, writeFileSync } from "fs";
+
+export class Dictionary implements IDictonary {
+  language: string;
+  wordList: DictPair[];
+  dictionaryFile: string;
+
+  constructor(dictionaryFile: string) {
+    this.dictionaryFile = dictionaryFile;
+    if (!existsSync(dictionaryFile)) {
+      throw new Error(`Could not open file: "${this.dictionaryFile}"`);
+    }
+    const d = JSON.parse(
+      readFileSync(this.dictionaryFile, "utf8")
+    ) as IDictonary;
+    this.language = d.language;
+    this.wordList = d.wordList;
+  }
+
+  toJSON(): unknown {
+    return {
+      language: this.language,
+      wordList: this.wordList,
+    };
+  }
+
+  saveDictionary(): void {
+    const spaces = 2;
+    writeFileSync(
+      this.dictionaryFile,
+      JSON.stringify(this.toJSON(), null, spaces),
+      "utf8"
+    );
+  }
+
+  addWord(word: string, replaceWith: string): Dictionary {
+    if (!this.exists(word)) {
+      const newWord: DictPair = {
+        word: word,
+        replacement: replaceWith,
+        settings: this.defaultSetting(),
+      };
+      this.wordList.push(newWord);
+      this.saveDictionary();
+    }
+    return this;
+  }
+
+  deleteWord(word: string): Dictionary {
+    this.wordList = this.wordList.filter((w) => w.word !== word);
+    this.saveDictionary();
+    return this;
+  }
+
+  find(word: string): DictPair | undefined {
+    return this.wordList.find(
+      (w) => w.word.toLocaleLowerCase() === word.toLocaleLowerCase()
+    );
+  }
+
+  searchAndReplace(text: string): string {
+    this.wordList.forEach((word) => {
+      word.settings = this.defaultSetting(word.settings);
+      const flags = word.settings.matchCasing ? "" : "i";
+      const re = new RegExp(`\\b${word.word}\\b`, flags);
+      const match = text.match(re);
+      if (match) {
+        const replaceValue = word.settings.keepCasingOnFirstCharacter
+          ? Dictionary.keepCasingOnFirstChar(match[0], word.replacement)
+          : word.replacement;
+        text = text.replace(re, replaceValue);
+      }
+    });
+    return text;
+  }
+
+  translateList(wordList: string[]): string[] {
+    return wordList.map((word): string => {
+      return this.translate(word);
+    });
+  }
+
+  translate(word: string): string {
```

Comment:
Shouldn't this be removed, and `searchAndReplace()` used instead?

Reply:
This `translate` and `translateList` was ideas that came up when I wrote the class. Intended as "might be useful" features. `translate` is intended as a simple function to translate a single word. `searchAndReplace` runs the entire dictionary against a text. With that said, there's no use in pushing unused features just for the sake of it.

 I'm not emotionally attached to these. Remove yes?

Reply:
I could see it as a useful feature for a dictionary - but have a hard time finding a use case for it in this extension... 

So if you don't have a future useful use case - remove!